client-v2/src/actions/FrontsConfig.js

```diff
@@ -28,6 +45,79 @@ function errorReceivingConfig(error: string): Action {
   };
 }
 
+function getFrontCollection(collectionId: string) {
+  return (dispatch: Dispatch, getState: () => State) =>
+    getCollection(collectionId)
+      .then((res: Object) => {
+        const collectionConfig = getCollectionConfig(getState(), collectionId);
```

Comment:
Why are there selectors in an action? Would it not be sending the raw data back to the reducer and deferring the derivations until they get used in the view?

Reply:
(me and Rich discussed this - we're happy it's necessary to do the necessary munging before our final collection type is added to the app state)